chore: prep release 1.0.0 (#755)
SwissDataScienceCenter/renku-notebooks: README.rst

    :target: https://conventionalcommits.org

> A simple service around the `Amalthea operator
> <https://github.com/SwissDataScienceCenter/amalthea>`_, which provides interactive Jupyter
> notebooks for the Renku platform.

Reviewer: ```suggestion
A simple service using the `Amalthea operator
<https://github.com/SwissDataScienceCenter/amalthea>`_, to provide interactive Jupyter
notebooks for the Renku platform.
```

The service relies on `renku-gateway <https://github.com/SwissDataScienceCenter/renku-gateway>`_
for authentication. However, anonymous users are supported as well in which case anyone can
start and use sessions for public renku projects. Therefore, the notebook service can run
even without having the renku-gateway installed or present. In this case only sessions 
for anonymous users can be launched.

Endpoints
---------

The service defines endpoints to list the active sessions for a user,

> start or stop a session. It can also provide information the logs from a running
> user sessions or information about work that was saved automatically if a user
> stops a sessions without commiting and pushing all their work to their project 

Reviewer: ```suggestion
start or stop a session. It can also provide the logs of a running
user session or information about work that was saved automatically if a user
stops a session without committing and pushing all their work to their project 
```

This will allow you to access the notebooks api and sessions without having to setup 
port forwarding with the ``kubectl`` command.

.. code-block:: console

    cat <<EOF | kind create cluster --name kind --config=-
    kind: Cluster
    apiVersion: kind.x-k8s.io/v1alpha4
    nodes:
    - role: control-plane
      kubeadmConfigPatches:
      - |
        kind: InitConfiguration
        nodeRegistration:
          kubeletExtraArgs:
            node-labels: "ingress-ready=true"
      extraPortMappings:
      - containerPort: 80
        hostPort: 80
        protocol: TCP
      - containerPort: 443
        hostPort: 443
        protocol: TCP
    EOF

2. Install ``ingress-nginx`` and the notebook service helm chart in the kind cluster.

.. code-block:: console

    $ VERSION=$(curl https://raw.githubusercontent.com/kubernetes/ingress-nginx/master/stable.txt)
    $ kubectl apply -f https://raw.githubusercontent.com/kubernetes/ingress-nginx/$VERSION/deploy/static/provider/kind/deploy.yaml
    $ cd helm-chart
    $ helm dep update ./renku-notebooks
    $ helm upgrade --install renku-notebooks ./renku-notebooks \
      --set "global.anonymousSessions.enabled=true" \

>       --set "gitlab.url=https://renkulab.io/gitlab" \

Reviewer 2: This makes me think, the gitlab URL and the registry URL should be passed in with the request at some point in the future, then one instance of the notebook service would work with multiple gitlabs

Author: Yeah that is pretty much the case. Maybe a few minor changes here and there are needed but not that many.

I think the biggest thing we have to solve for multiple gitlabs is the git proxy perhaps.

      --set "gitlab.registry.host=registry.renkulab.io" \

>       --set "amalthea.scope.namespaces[0]=default" \

Reviewer 2: Are you actually sure that this works? You can set an entire array through the helm command line, see https://github.com/helm/helm/issues/1987#issuecomment-280497496

Author: Yeah I just went through the instructions again. And I can successfully install the helm chart with that command. Here are the relevant sections of running `helm get values` after installing.

```yaml
USER-SUPPLIED VALUES:
amalthea:
  scope:
    namespaces:
    - default
 ...
ingress:
  annotations:
    kubernetes.io/ingress.class: nginx
  enabled: true
  hosts:
  - localhost
```
